pkg/server/handler.go

```diff
@@ -343,7 +343,16 @@ func mergeCRDsIntoCoreGroup(crdLister kcp.ClusterAwareCRDLister, crdHandler, cor
 			// server handle it.
 			crdName := requestInfo.Resource + ".core"
 
-			if _, err := crdLister.Get(req.Request.Context(), crdName); err == nil {
+			clusterName, err := request.ClusterNameFrom(req.Request.Context())
+			if err != nil {
+				responsewriters.ErrorNegotiated(
+					apierrors.NewInternalError(fmt.Errorf("no cluster found in the context")),
+					// TODO is this the right Codecs?
```

Comment:
rm?

Reply:
Copy-pasta from the rest of the file, keeping for consistency